docker-compose.yml

```diff
@@ -0,0 +1,41 @@
+version: '2'
+
+services:
+    nginx:
+        image: nginx:latest
+        container_name: cqrs_codely_tv_nginx
+        ports:
+            - "80:80"
+        volumes:
+            - .:/app
+            - ./etc/infrastructure/nginx/site.conf:/etc/nginx/conf.d/default.conf
+        depends_on:
+            - php
+        volumes_from:
+            - php
+        links:
+            - php
+
+    php:
+        image: jorge07/alpine-php:7.1-dev
+        container_name: cqrs_codely_tv_php
+        ports:
+            - "2244:22"
+            - "9000:9000"
+        volumes:
+            - "$PWD:/app"
+        environment:
+            - PHP_IDE_CONFIG=serverName=CodelyCQRSCourse
+        depends_on:
+            - mysql
+
+    mysql:
+        image: mysql:5.7
+        container_name: cqrs_codely_tv_mysql
+        ports:
+            - "3306:3306"
+        environment:
+            MYSQL_ROOT_PASSWORD: root
+            MYSQL_DATABASE: cqrs-codely-tv
+            MYSQL_USER: cqrscodelytv
+            MYSQL_PASSWORD: cqrscodelytv
```

Comment:
We should avoid hardcoding environment dependent values in the Docker compose file. The reason is because this way we would be making the application users to set these MySQL root user password, use this DB name, and so on. Instead, we can just pass on an environment variable to the container:

How would it looks like?

`docker-compose.yml`:
```yml
    mysql:
        image: mysql:5.7
        container_name: cqrs_codely_tv_mysql
        ports:
            - "3306:3306"
        environment:
            MYSQL_ROOT_PASSWORD: ${CODELYTV_CQRS_MYSQL_ROOT_PASSWORD}
            MYSQL_DATABASE:  ${CODELYTV_CQRS_MYSQL_DATABASE}
            MYSQL_USER: ${CODELYTV_CQRS_MYSQL_USER}
            MYSQL_PASSWORD: ${CODELYTV_CQRS_MYSQL_PASSWORD}
```

How to set these new environment variables on the host machine?

* Option 1: Set them on the host machine with: `export CODELYTV_CQRS_MYSQL_USER=lerele`
* Option 2: Specify them while running the Docker compose with: `docker-compose run -e CODELYTV_CQRS_MYSQL_USER=lerele`
* Option 3: Environment file `.env`:
```bash
CODELYTV_CQRS_MYSQL_USER=lerele
```
* Option 4: `env_file` configuration option: `docker run --env-file=FILE`

Here you have [more information on that technics](https://docs.docker.com/compose/environment-variables/#the-env-file).

In my opinion, we could go for the `.env` file in order to do not have to specify too much configuration options while running the containers. However, in order to avoid versioning these details, I would propose the following approach:
1. Create an `.env.dist` file with all the needed environment variables with default values
1. Ignore the `.env` file through the `.gitignore` in order to avoid versioning these details (they will be different from user to user)
1. Specify in the `README.md` how to setup the application (just copy the `.env.dist` file naming it `.env` and specifying the desired values for each environment variable).

What do you think about this proposal?